api/config/v1alpha1/envoygateway_types.go

```diff
@@ -86,6 +94,46 @@ type FileProvider struct {
 	// TODO: Add config as use cases are better understood.
 }
 
+// RateLimit defines the configuration associated with the Rate Limit Service
+// using for Global Rate Limiting.
+type RateLimit struct {
+	// Backend holds the configuration associated with the
+	// database backend used by the rate limit service to store
+	// state associated with global ratelimiting.
+	Backend *RateLimitDatabaseBackend `json:"backend"`
+}
+
+// RateLimitDatabaseBackend defines the configuration associated with
+// the database backend used by the rate limit service.
+type RateLimitDatabaseBackend struct {
+	// Type is the type of database backend to use. Supported types are:
+	//	* Redis: Connects to a Redis database.
+	//
+	// +unionDiscriminator
+	Type RateLimitDatabaseBackendType `json:"type"`
+	// Redis defines the settings needed to connect to a Redis database.
+	//
+	// +optional
+	Redis *RateLimitRedisSettings `json:"redis,omitempty"`
+}
+
+// RateLimitDatabaseBackendType specifies the types of database backend
+// to be used by the rate limit service.
+// +kubebuilder:validation:Enum=Redis
+type RateLimitDatabaseBackendType string
+
+const (
+	// RedisBackendType uses a redis database for the rate limit service.
+	RedisBackendType RateLimitDatabaseBackendType = "Redis"
+)
+
+// RateLimitRedisSettings defines the configuration for connecting to
+// a Redis database.
+type RateLimitRedisSettings struct {
```

Comment:
Why is this a string pointer? This seems liek a required field. If not, keey the pointer and add a godoc that states the default value when unspecified and the appropriate kubebuilder `// +optional` tag.

Reply:
it is a required field, pointers help differentiate between nil and empty strings, not very useful for this case, can move it to `string` if that is your preference

Reply:
Please do.